gobblin-yarn/src/main/java/gobblin/yarn/YarnService.java

```diff
@@ -425,12 +506,9 @@ public void onContainerStarted(ContainerId containerId, Map<String, ByteBuffer>
     @Override
     public void onContainerStatusReceived(ContainerId containerId, ContainerStatus containerStatus) {
       LOGGER.info(String.format("Received container status for container %s: %s", containerId, containerStatus));
-      if (containerStatus.getState() == ContainerState.COMPLETE) {
-        LOGGER.info(String.format("Container %s completed with status %s", containerId, containerStatus));
-        containerMap.remove(containerId);
 
-        LOGGER.info("Requesting a new container to replace the one that has completed");
-        eventBus.post(new NewContainerRequest(1));
+      if (containerStatus.getState() == ContainerState.COMPLETE) {
+        handleContainerCompletion(containerStatus);
```

Comment:
Actually I'm still not sure which one or both callback methods will be called in case a container completes. We can keep the one in `onContainersCompleted`.